### ISSUE TITLE
Error while installing 

### ISSUE THREAD

**Issue by RobinMalfait** [LinusU/node-appdmg#68]
Hey Guys

When I try `npm install -g appdmg` I get the following error:

```
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/xattr.node] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/usr/local/bin/iojs" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/appdmg/node_modules/fs-xattr
gyp ERR! node -v v3.3.0
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/volume.node] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 15.0.0
gyp ERR! command "/usr/local/bin/iojs" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /usr/local/lib/node_modules/appdmg/node_modules/ds-store/node_modules/macos-alias
gyp ERR! node -v v3.3.0
gyp ERR! node-gyp -v v2.0.2
gyp ERR! not ok
npm ERR! Darwin 15.0.0
npm ERR! argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "install" "-g" "appdmg"
npm ERR! node v3.3.0
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! fs-xattr@0.1.11 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fs-xattr@0.1.11 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the fs-xattr package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls fs-xattr
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/robin/Tools/Claever/ClaRoDesktopApplicatie/npm-debug.log
```

Any ideas?

**Comment by RobinMalfait** [comment]
I've tried reinstalling GCC, but didn't work

**Comment by felicienfrancois** [comment]
It may be an issue with xcode. Some gyp build require xcode to be installed (not only the command line tools, the full one). Additionally, some people have issues with old xcode on newer osx. You should install the latest xcode (xcode 7).

According to [this thread](https://github.com/Homebrew/homebrew/issues/40653), you could also fix this issue by creating a symlink (but you may encounter other issue if you don't fix your xcode)

**Comment by RobinMalfait** [comment]
I'm on El Capitan with xcode 6.1, I'll try to update xcode :) thanks @felicienfrancois

**Comment by LinusU** [comment]
Upgrade Xcode and run `xcode-select --install` solved it for me, please reopen if issue persists :)

**Comment by RobinMalfait** [comment]
@LinusU updating xcode worked :D

**Comment by LinusU** [comment]
:ok_hand:

**Comment by ccnokes** [comment]
When trying to install electron-builder, it always fails on this module, on these dependencies: fs-xattr@0.1.11 and macos-alias@0.2.9. They both error out with "ld: library not found for -lgcc_s.10.5". 

I have wine-1.6.2 and makensis v05-Oct-2015.cvs installed via brew and Xcode 6.4 running (I also ran the `xcode-select --install` the command above). I'm also on El Capitan. Any ideas on what's going on? Let me know if you need more details.

**Comment by LinusU** [comment]
Did you select "Install command line tools" when you ran `xcode-select --install`?

**Comment by ccnokes** [comment]
I didn't have to select anything, it just started installing something. Now it says I'm running `xcode-select version 2343.`. Do I need to reinstall that maybe?

**Comment by LinusU** [comment]
Hmm, I actually don't know, can you compile anything on your system?

Try creating the following file, `test.c`, and run `clang test.c`.

```c
int main () {
  return 0;
}
```

**Comment by ccnokes** [comment]
That seemed to work (I don't know much about C), it outputted a `a.out` file. Is that right?

**Comment by LinusU** [comment]
Yes, that's right... Could you try cloning this repo and running `npm install`?

**Comment by ccnokes** [comment]
So I get the same errors on the same modules. I thought I'd include more of the output from those here just in case it helps:
```
ld: library not found for -lgcc_s.10.5
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Release/volume.node] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2` 
```

I also get errors on capture-window@0.1.3, where it outputs: 
```
  CXX(target) Release/obj.target/capture-window/src/capture-window.o
In file included from ../src/capture-window.cc:8:
In file included from ../src/apple.m:2:
In file included from /System/Library/Frameworks/Cocoa.framework/Headers/Cocoa.h:12:
In file included from /System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:10:
In file included from /System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:6:
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:7:15: error: expected ';' after @class
@class NSArray<ObjectType>;
```
And then has many more errors in the same file. The npm-debug.log is 14k lines. I can send that to you if you like. 

(Thanks for helping me with this btw)

**Comment by ccnokes** [comment]
So I just reinstalled Xcode and it's working. The latest version is 7.0.1, I was on 6.4 previously (I thought that was the latest, sorry). Everything's working. 

I had also removed `xcode-select` via `rm -rf /Library/Developer/CommandLineTools`. Reinstalling Xcode seems to have automatically installed `xcode-select` as well (same version, 2343). It looks like my gcc was upgraded as well from 6.1.0 to `Apple LLVM version 7.0.0 (clang-700.0.72)`. Anyways, thanks for your help!
